src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -151,6 +152,23 @@ public IIntellisenseResult Suggest(string expression, RecordType parameterType,
             return suggestions;
         }
 
+        /// <summary>
+        /// Creates a renamer instance for updating a field reference from <paramref name="parameters"/> in expressions.
+        /// </summary>
+        /// <param name="parameters">Type of parameters for formula. The fields in the parameter record can 
+        /// be acecssed as top-level identifiers in the formula. Must be the names from before any rename operation is applied.</param>
+        /// <param name="pathToRename">Path to the field to rename.</param>
+        /// <param name="updatedName">New name. Replaces the last segment of <paramref name="pathToRename"/>.</param>
+        /// <returns></returns>
+        public RenameDriver CreateFieldRenamer(RecordType parameters, DPath pathToRename, DName updatedName)
+        {
+            Contracts.AssertValue(parameters);
```

Comment:
> Contracts.AssertValue [](http://example.com/codeflow?start=11&length=22)

Are these debug-only checks? Since this is a public method, we want to check them in release.